File: csharp/HearthstoneReplays/Parser/Node.cs
#region

using System;

#endregion

namespace HearthstoneReplays.Parser
{
	public class Node
	{
		public Node(Type type, object o, int indentLevel, Node parent)
		{
			Type = type;
			Object = o;
			IndentLevel = indentLevel;
			Parent = parent;
		}

		public Type Type { get; set; }
		public object Object { get; set; }
		public int IndentLevel { get; set; }
		public Node Parent { get; set; }
	}
}

File: java/src/main/java/info/hearthsim/hsreplay/parser/Node.java
package info.hearthsim.hsreplay.parser;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@ToString(of = { "type", "parent" })
public class Node {

	private Class<? extends Object> type;
	private Object object;
	private int indentLevel;
	private Node parent;
}
